src/HDF5.jl

```diff
@@ -1752,23 +1752,35 @@ write(parent::Union{HDF5File, HDF5Group}, name::String, data::Union{T, AbstractA
 # For datasets, "write(dset, name, val)" means "a_write"
 write(parent::HDF5Dataset, name::String, data::Union{T, AbstractArray{T}}, plists...) where {T<:ScalarOrString} = a_write(parent, name, data, plists...)
 
-# Reading arrays using getindex: data = dset[:,:,10]
-function getindex(dset::HDF5Dataset, indices::Union{AbstractRange{Int},Int}...)
+
+# Indexing
+
+Array(dset::HDF5Dataset) = read(dset)
```

Comment:
Unrelated? Why add this?

Reply:
I'll put it it another PR for clarity.

Reply:
did you forget to remove this for this PR?

Reply:
must have undone it and rebased it back in.